xmtp_id/src/associations/signature.rs

```diff
@@ -363,6 +379,37 @@ impl Signature for LegacyDelegatedSignature {
     }
 }
 
+/// Decode the `legacy_signed_private_key` to legacy private / public key pairs & sign the `signature_text` with theprivate key.
+pub async fn sign_with_legacy_key(
+    signature_text: String,
+    legacy_signed_private_key: Vec<u8>,
+) -> Result<LegacyDelegatedSignature, SignatureError> {
+    let legacy_signed_private_key_proto =
+        LegacySignedPrivateKeyProto::decode(legacy_signed_private_key.as_slice())?;
+    let signed_private_key::Union::Secp256k1(secp256k1) = legacy_signed_private_key_proto
+        .union
+        .ok_or(SignatureError::MalformedLegacyKey(
+            "Missing secp256k1.union field".to_string(),
+        ))?;
+    let legacy_private_key = secp256k1.bytes;
+    let wallet: LocalWallet = hex::encode(legacy_private_key).parse::<LocalWallet>()?;
+    let signature = wallet.sign_message(signature_text.clone()).await?;
+
```

Comment:
This is the old logic:

```
    let legacy_private_key = secp256k1.bytes;
    let (mut delegating_signature, recovery_id) = k256_helper::sign_sha256(
        &legacy_private_key, // secret_key
        // TODO: Verify this will create a verifiable signature
        signature_text.as_bytes(), // message
    )
    .map_err(IdentityError::LegacySignature)?;
    delegating_signature.push(recovery_id); // TODO: normalize recovery ID if necessary
 ```
  Out of curiosity, what is the reason we need to change this to construct a `LocalWallet` out of the `legacy_private_key` here?

Reply:
Because `LocalWallet` wraps the signature_text with EIP-191 text while old logic doesn't. EIP-191 is implemented by evm wallet like Metamask. If it's not use, we won't be able to recover the signature to the signer address properly.

We can add the EIP-191 to the old logic, but I feel it's better to abstract away EIP-191 from our codebase.

Reply:
[ethers crate source code](https://docs.rs/ethers-signers/2.0.14/src/ethers_signers/wallet/mod.rs.html#85) to add EIP-191 during signing, note the `hash_message`.

Reply:
Ahhh that explains a lot, thank you!